### PR TITLE
Brings support of docker collection

### PR DESCRIPTION
**Pull request by thibaultzanini**

### CODE REVIEW

File: core/dao/docker-collection-dao.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+var AbstractDao = require("core/dao/abstract-dao").AbstractDao,
+    Model = require("core/model/model").Model;
+
+exports.DockerCollectionDao = AbstractDao.specialize({
+    init: {
+        value: function() {
+            this._model = Model.DockerCollection;
```
Comment (pchaussalet) [inline core/dao/docker-collection-dao.js]: `Model` is available at `this.constructor.Model`, no need to require it there.

File: core/dao/docker-container-creator-dao.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+var AbstractDao = require("core/dao/abstract-dao").AbstractDao,
+    Model = require("core/model/model").Model;
+
+exports.DockerContainerCreatorDao = AbstractDao.specialize({
+    init: {
+        value: function() {
+            this._model = Model.DockerContainerCreator;
```
Comment (pchaussalet) [inline core/dao/docker-container-creator-dao.js]: `Model` is available at `this.constructor.Model`, no need to require it there.

File: ui/sections/containers/controls/docker-collection-list.reel/docker-collection-list.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            return Promise.all([
+                this._sectionService.getNewDockerCollection(),
+                this._sectionService.listDockerCollection(),
+                this._sectionService.getNewDockerContainerCreator()
+            ]).then(function (data) {
+                var dockerCollection = data[0],
+                    dockerCollections = data[1];
+
+                //Default Collection?
+                dockerCollection.id = -1;
```
Comment (pchaussalet) [inline ui/sections/containers/controls/docker-collection-list.reel/docker-collection-list.js]: You should move that to `container-repository`.

File: ui/sections/containers/inspectors/container.reel/container.js [hunk 4]
```diff
@@ -61,6 +80,23 @@ exports.Container = AbstractInspector.specialize({
                 window.open(this.object.web_ui_url);
             }
         }
+    },
+
+    _mapObjectEnvironments: {
```
Comment (pchaussalet) [inline ui/sections/containers/inspectors/container.reel/container.js:85]: You should move this to `container-section-service`.

**Review by pchaussalet — CHANGES_REQUESTED** [review]

File: ui/sections/containers/controls/docker-collection-list.reel/docker-collection-list.js (hunk 3 above)
Comment (thibaultzanini) [reply ui/sections/containers/controls/docker-collection-list.reel/docker-collection-list.js]: True

File: ui/sections/containers/inspectors/container.reel/container.js (hunk 4 above)
Comment (thibaultzanini) [reply ui/sections/containers/inspectors/container.reel/container.js:85]: It's really specific to the view,

**Comment by thibaultzanini** [comment]
PR updated.

**Review by pchaussalet — APPROVED** [review]
LGTM

**Pull request merged**